test/statistics_test.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (C) 2022 by Daniel Shapero <shapero@uw.edu>
+#
+# This file is part of icepack.
+#
+# icepack is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# The full text of the license can be found in the file LICENSE in the
+# icepack source directory or at <http://www.gnu.org/licenses/>.
+
+import pytest
+import numpy as np
+import numpy.random as random
+import firedrake
+from firedrake import inner, grad, dx, exp, interpolate, as_vector
+import icepack
+from icepack.statistics import StatisticsProblem, MaximumProbabilityEstimator
+from icepack.constants import (
+    ice_density as ρ_I,
+    water_density as ρ_W,
+    gravity as g,
+    glen_flow_law as n,
+)
+
+
+def test_poisson_problem():
+    Nx, Ny = 32, 32
+
+    mesh = firedrake.UnitSquareMesh(Nx, Ny)
+    degree = 2
+    Q = firedrake.FunctionSpace(mesh, "CG", degree)
+    V = firedrake.FunctionSpace(mesh, "CG", degree)
+
+    x, y = firedrake.SpatialCoordinate(mesh)
+    f = interpolate(firedrake.Constant(1), Q)
+
+    def simulation(q):
+        u = firedrake.Function(V)
+        J = (0.5 * exp(q) * inner(grad(u), grad(u)) - f * u) * dx
+        F = firedrake.derivative(J, u)
+
+        bc = firedrake.DirichletBC(V, 0, "on_boundary")
+        problem = firedrake.NonlinearVariationalProblem(F, u, bc)
+        params = {
+            "solver_parameters": {
+                "pc_type": "lu",
+                "pc_factor_mat_solver_type": "mumps",
+            }
+        }
+        solver = firedrake.NonlinearVariationalSolver(problem, **params)
+        solver.solve()
```

Comment:
My firedrake and python are feeling rusty - out of interest is there any advantage to doing this explicitly instead of just calling `firedrake.solve(F==0, u, bc, **params)`?